src/main.rs

```diff
@@ -87,168 +82,120 @@ async fn floodfill_pt(data: web::Data<AppState>, input: web::Json<UserInputJSON>
-        let graph_pt_guard = arc_graph_pt.lock().unwrap();
-        let node_values_1d_guard = arc_node_values_1d.lock().unwrap();
+    let count_original_nodes = graph_walk.len() as u32;
 
-        let count_original_nodes: u32 = graph_walk_guard.len() as u32;
-
-        let graph_walk_unguarded = &*graph_walk_guard;
-        let graph_pt_unguarded = &*graph_pt_guard;
-        let node_values_1d_unguarded = &*node_values_1d_guard;
-
-        println!(
-            "Creating tuples to pass to floodfill for {} data",
-            input.year
-        );
-        for i in 0..input.start_nodes_user_input.len() {
-            model_parameters_each_start.push((
-                graph_walk_unguarded,
-                NodeID(input.start_nodes_user_input[i] as u32),
-                node_values_1d_unguarded,
+    println!(
+        "Started running floodfill\ttime_of_day_ix: {}\tNodes count: {}",
+        time_of_day_ix,
+        input.start_nodes_user_input.len()
+    );
+    let now = Instant::now();
+    let indices = (0..input.start_nodes_user_input.len()).collect::<Vec<_>>();
+    let results: Vec<(i32, u32, [i64; 32], Vec<u32>, Vec<u16>)> = indices
+        .par_iter()
+        .map(|i| {
+            floodfill(
+                &graph_walk,
+                &graph_pt,
+                NodeID(input.start_nodes_user_input[*i] as u32),
+                &node_values_1d,
                 &data.travel_time_relationships_all[time_of_day_ix],
                 &data.subpurpose_purpose_lookup,
-                graph_pt_unguarded,
                 input.trip_start_seconds,
-                Cost(input.init_travel_times_user_input[i] as u16),
+                Cost(input.init_travel_times_user_input[*i] as u16),
                 count_original_nodes,
-                node_values_padding_row_count,
+                data.node_values_padding_row_count,
                 &input.target_destinations,
-            ))
-        }
+            )
+        })
+        .collect();
+    println!("Parallel floodfill took {:?}", now.elapsed());
 
-        println!(
-            "Started running floodfill\ttime_of_day_ix: {}\tNodes count: {}",
-            time_of_day_ix,
-            model_parameters_each_start.len()
-        );
-        let now = Instant::now();
-        parallel_res = model_parameters_each_start
-            .par_iter()
-            .map(|input| floodfill(*input))
-            .collect();
-        println!("Parallel floodfill took {:?}", now.elapsed());
-    } else {
-        let count_original_nodes: u32 = graph_walk_guard.len() as u32;
+    serde_json::to_string(&results).unwrap()
+}
+
+// No changes to the graph allowed
+fn floodfill_pt_no_changes(data: web::Data<AppState>, input: web::Json<UserInputJSON>) -> String {
+    println!("Floodfill request received, without changes");
+    let time_of_day_ix = get_time_of_day_index(input.trip_start_seconds);
+
+    let (node_values_1d, graph_walk, graph_pt, node_values_padding_row_count) =
```

Comment:
In this case, all the previous arc/mutex stuff here was meaningless. These're loaded and scoped to each request.